lib/match.test.js

```diff
@@ -139,26 +139,56 @@ describe("match", function() {
         assert.isFalse(checkMatch);
     });
 
+    it("returns false if null and empty object", function() {
+        var checkMatch = samsam.match(null, {});
+        assert.isFalse(checkMatch);
+    });
+
+    it("returns false if null and object with properties", function() {
+        var checkMatch = samsam.match(null, { not: 1 });
+        assert.isFalse(checkMatch);
+    });
+
     it("returns false if undefined and empty string", function() {
         var checkMatch = samsam.match(undefined, "");
         assert.isFalse(checkMatch);
     });
 
+    it("returns false if undefined and empty object", function() {
+        var checkMatch = samsam.match(undefined, {});
+        assert.isFalse(checkMatch);
+    });
+
     it("returns false if false and empty string", function() {
         var checkMatch = samsam.match(false, "");
         assert.isFalse(checkMatch);
     });
 
+    it("returns false if false and empty object", function() {
+        var checkMatch = samsam.match(false, {});
+        assert.isFalse(checkMatch);
+    });
+
     it("returns false if 0 and empty string", function() {
         var checkMatch = samsam.match(0, "");
         assert.isFalse(checkMatch);
     });
 
+    it("returns false if 0 and empty object", function() {
+        var checkMatch = samsam.match(0, {});
+        assert.isFalse(checkMatch);
+    });
+
     it("returns false if NaN and empty string", function() {
         var checkMatch = samsam.match(NaN, "");
         assert.isFalse(checkMatch);
     });
 
+    it("returns false if NaN and empty object", function() {
+        var checkMatch = samsam.match(NaN, {});
+        assert.isFalse(checkMatch);
+    });
+
```

Comment:
```suggestion

    it("returns false if RegExp and empty object", function() {
        var checkMatch = samsam.match(/\w/, {});
        assert.isFalse(checkMatch);
    });
```
The above fails, do you think this is worth considered?

Reply:
Interesting question. I was just asking myself if matching against an empty object makes any sense at all. However, it could be seen as "match any object". Seen this way matching a regular express again `{}` should return `true`. This is in line with the current behavior for `samsam.match([], {})` and `samsam.match({}, [])`. Those also return `true`.

In your example, as soon as you're adding a property to the object, the return value is `false`, e.g. `samsam.match(/\w/, { not: 1 })`.

One alternative could be to treat an empty object matcher as an error and throw an exception. I'm not sure about this.

Reply:
Thanks for your explanation and feedback